### PR TITLE
Fix time check for dataset with timeseries of variable ranges

### PR DESCRIPTION
**Pull request by lixiliu**
Not done, still need to test and validate most of the code changes.

Completes JIRA Story DSGRID-
Closes GitHub Issue #

## Checklist

- [ ] Tests exercising the new feature or bug fix
- [ ] All tests pass
- [ ] At least one code review approval
- [ ] Consider transferring TODOs to JIRA or GitHub

### CODE REVIEW

File: dsgrid/registry/dataset_registry_manager.py [hunk 1]
```diff
@@ -129,17 +131,78 @@ def _check_lookup_data_consistency(self, config: DatasetConfig, load_data_lookup
...
+                "id",
+                "left",
+            ).select(
+                "id",
+                "geography",
+                "timestamp",
+                F.lit(data_tz).alias("from_timezone"),
+                F.col("to_id").alias("to_timezone"),
+            )
+            udf_find_time_delta = F.udf(lambda fm, to: find_time_delta(fm, to), T.DateType())
```
Comment (daniel-thom) [inline dsgrid/registry/dataset_registry_manager.py:169]: I don't think I'm following the type you are creating here. The find_time_delta function returns a Python timedelta object and I don't see how that gets stored in a DateType. Do you want to convert the timedelta to an integer expressed as number of seconds?

Also, note that UDFs in Python are very slow. There may not be a better way.

Comment (lixiliu) [reply dsgrid/registry/dataset_registry_manager.py:169]: You are right, I did not look super hard, maybe `CalendarIntervalType` is more appropriate? https://sparkbyexamples.com/pyspark/pyspark-sql-types-datatype-with-examples/

Comment (daniel-thom) [reply dsgrid/registry/dataset_registry_manager.py:169]: I'm not sure that that is actually a Spark type. https://spark.apache.org/docs/latest/sql-ref-datatypes.html

`timedelta.total_seconds()` stored as an integer is probably good enough.

File: dsgrid/config/dimensions.py [hunk 2]
```diff
@@ -339,7 +339,22 @@ class TimeDimensionModel(TimeDimensionBaseModel):
...
+        description="""
+        Time zone of the ranges specified:
+            UTC, 
+            HawaiiAleutianStandard, 
+            AlaskaStandard, AlaskaPrevailing,
+            PacificStandard, PacificPrevailing, 
+            MountainStandard, MountainPrevailing, 
+            CentralStandard, CentralPrevailing, 
+            EasternStandard, EasternPrevailing,
+            LOCAL
```
Comment (mooneyme) [inline dsgrid/config/dimensions.py:355]: Is ComStock/ResStock data in LOCAL prevailing? Or LOCAL standard?

Comment (mooneyme) [inline dsgrid/config/dimensions.py:355]: This should probably go into an enumeration instead of being duplicated here in the description.

Comment (mooneyme) [reply dsgrid/config/dimensions.py:355]: Looks like you don't need to add any of these. The docs take care of it:
image

**Review by mooneyme — CHANGES_REQUESTED** [review]
@lixiliu  can you please add a test for these new changes?

File: dsgrid/dimension/time.py [hunk 3]
```diff
@@ -94,22 +94,36 @@ class MeasurementType(DSGEnum):
...
+    America/Denver (MPT)
+    America/Chicago (CPT)
+    America/New_York (EPT)
+    """
 
     UTC = EnumValue(
         value="UTC",
         description="Coordinated Universal Time",
-        tz=datetime.timezone(datetime.timedelta()),
+        tz=pytz.timezone("UTC"),  # datetime.timezone(datetime.timedelta()),
```
Comment (daniel-thom) [inline dsgrid/dimension/time.py:116]: Remove the old code?

File: dsgrid/dimension/time.py [hunk 4]
```diff
@@ -259,3 +273,16 @@ def make_time_range(start, end, frequency, leap_day_adjustment):
             "366 days not allowed for frequency, use 365 days to specify annual frequency."
         )
     return DatetimeRange(start, end, frequency, leap_day_adjustment)
+
+
+def find_time_delta(timestamp, from_tz: str, to_tz: str):
+    """
+    find datetime.timedelta (in seconds) between two time zones
+    output: integer (sec)
```
Comment (daniel-thom) [inline dsgrid/dimension/time.py:281]: numpy docstring format
```
Returns
--------
int
    Difference between the two time zones in seconds.
```

File: dsgrid/config/dimensions.py (hunk 2 above)
Comment (lixiliu) [reply dsgrid/config/dimensions.py]: LOCAL standard

**Pull request closed without merging**